solv/simplequery.c

```diff
@@ -132,3 +132,25 @@ SolvFindSolvablesByNevraStr(
 error:
     goto cleanup;
 }
+
+uint32_t
+SolvRequiresFromQueue(
+    Pool *pool,
+    Queue *pq_pkgs,  /* solvable ids */
+    Queue *pq_deps   /* string ids */
+)
+{
+    uint32_t dwError = 0;
+    int i,j;
+
+    for (i = 0; i < pq_pkgs->count; i++) {
+        Queue q_tmp = {0};
+        Solvable *p_solv = pool_id2solvable(pool, pq_pkgs->elements[i]);
```

Comment:
validate the return values, generic comment

Reply:
Look at the code in `libsolv`. They never do that. `pool_id2solvable` is actually an inline function that won't return any error.